test/source/coiteration_test.cpp

```diff
@@ -193,3 +189,82 @@ TEST_CASE("Coiteration-Singleton-Singleton-Dense-Dense")
     }
     CHECK(fn(std::tuple(it1 == end1, it2 == end2, it3 == end3, it4 == end4)) == true);
 }
+
+TEST_CASE("Coiteration-Dense-Hashed-ConjunctiveMerge")
+{
+    /* Test coiteration for dense and hashed formats.
+
+    A conjunctive merge requires coiterating over a dense and hashed format. This
+    test checks that the coiteration is done correctly. The test proceeds as follows:
+
+    - If one of the levels is unordered (e.g. hashed, or singleton), then the
+      coiteration is done by iterating over the ordered level, and then looking up
+      the corresponding value in the unordered level.
+    - The coiteration stops when the end of the ordered (i.e. dense) level is reached.
+
+    This test checks that the lookup is done correctly.
+    */
+    constexpr uint8_t ZERO = 0;
+
+    std::unordered_map<uintptr_t, uintptr_t> const umap1{ { 0, 1 }, { 2, 0 }, { 1, 2 } };
+    std::vector<std::unordered_map<uintptr_t, uintptr_t>> const crd0{ umap1 };
+
+    // initialize the two levels to be coiterated
+    xsparse::levels::dense<std::tuple<>, uintptr_t, uintptr_t> dense_level{ 5 };
+    xsparse::levels::hashed<
+        std::tuple<>,
+        uintptr_t,
+        uintptr_t,
+        xsparse::util::container_traits<std::vector, std::unordered_set, std::unordered_map>,
+        xsparse::level_properties<false, false, false, false, false>>
+        hash_level{ 5, crd0 };
+
+    // define a conjunctive function
+    auto fn = [](std::tuple<bool, bool> t) constexpr { return (std::get<0>(t) && std::get<1>(t)); };
+
+    xsparse::level_capabilities::Coiterate<std::function<bool(std::tuple<bool, bool>)>,
+                                           uintptr_t,
+                                           uintptr_t,
+                                           std::tuple<decltype(dense_level), decltype(hash_level)>,
+                                           std::tuple<>>
+        coiter(fn, dense_level, hash_level);
+
+    // define iteration helper through dense and hashed level
+    auto it_helper1 = dense_level.iter_helper(std::make_tuple(), ZERO);
+    auto it_helper2 = hash_level.iter_helper(ZERO);
```

Comment:
The hashed level may iterate in a weird order -- Let's not assume we need the iterator, and check things directly using `locate` instead.

Reply:
Removed iter_helpers for the hashed level.